tasks/lex.rake

```diff
@@ -230,13 +230,83 @@ task "lex:rubygems": :compile do
   RESULTS
 end
 
+TOP_100_GEM_FILENAME = "tasks/top-100-gems.yml"
+TOP_100_GEMS_DIR = "top-100-gems"
+
+desc "Download the top 100 rubygems under #{TOP_100_GEMS_DIR}/"
+task "download:topgems" do
+  $:.unshift(File.expand_path("../lib", __dir__))
+  require "net/http"
+  require "rubygems/package"
+  require "tmpdir"
+
+  queue = Queue.new
+  YAML.safe_load_file(TOP_100_GEM_FILENAME).each do |gem_name|
+    gem_url = "https://rubygems.org/gems/#{gem_name}.gem"
+    queue << [gem_name, gem_url]
+  end
+
+  Dir.mkdir TOP_100_GEMS_DIR unless File.directory? TOP_100_GEMS_DIR
+
+  workers =
+    ENV.fetch("WORKERS", 16).times.map do
+      Thread.new do
+        Net::HTTP.start("rubygems.org", 443, use_ssl: true) do |http|
+          until queue.empty?
+            (gem_name, gem_url) = queue.shift
+            directory = File.expand_path("#{TOP_100_GEMS_DIR}/#{gem_name}")
+            unless File.directory? directory
+              puts "Downloading #{gem_name}"
+
+              http.request(Net::HTTP::Get.new(gem_url)) do |response|
+                # Skip unexpected responses
+                raise gem_url unless response.is_a?(Net::HTTPSuccess)
+
+                Dir.mktmpdir do |tmpdir|
+                  filepath = File.join(tmpdir, "#{gem_name}.gem")
+                  File.write(filepath, response.body)
+
+                  Gem::Package.new(filepath).extract_files(directory, "**/*.rb")
+                end
+              end
+            end
+          end
+        end
+      end
+    end
+
+  workers.each(&:join)
+end
+
+desc "Parse the top 100 rubygems"
+task "parse:topgems": ["download:topgems", :compile] do
+  require "yarp"
+  require_relative "../test/test_helper"
+
+  class ParseTop100GemsTest < Test::Unit::TestCase
+    Dir["#{TOP_100_GEMS_DIR}/**/*.rb"].each do |filepath|
+      test filepath do
+        result = YARP.parse_file_dup(filepath)
+        unless result.success?
+          if system(RbConfig.ruby, "-c", filepath, out: File::NULL, err: File::NULL)
+            raise "#{filepath} had errors but CRuby parsed it fine: #{result.errors}"
+          else
+            puts "\n#{filepath} was ignored as it is not valid syntax"
```

Comment:
What's the purpose of this `puts`? Is it helpful for us to know which files were ignored?